hail/python/hail/methods/statgen.py

```diff
@@ -456,8 +456,13 @@ def make_one_y_matrix(ys_and_covs_to_keep, one_y_field_name_set):
         return hl.nd.array(ys_and_covs_to_keep.map(lambda struct: array_from_struct(struct, one_y_field_name_set)))
 
     def setup_globals(ht):
-        list_of_ys_and_covs_to_keep_with_indices = \
-            [hl.enumerate(ht[sample_field_name]).filter(lambda struct_with_index: all_defined(struct_with_index[1], one_y_field_name_set + cov_field_names)) for one_y_field_name_set in y_field_names]
+        all_covs_defined = ht[sample_field_name].map(lambda sample_struct: all_defined(sample_struct, cov_field_names))
```

Comment:
I believe tests are failing because `cov_field_names` can be empty, so if you pass that to`all_defined` you hit that case where you'd need to do `hl.empty_array`.  This isn't a problem in the old version since there has to be at least one cov or phenotype.

Reply:
If you'd like I can just fix and push